python/cudf/cudf/core/series.py

```diff
@@ -815,6 +817,100 @@ def __array_function__(self, func, types, args, kwargs):
         else:
             return NotImplemented
 
+    def map(self, arg, na_action=None) -> "Series":
+        """
+        Map values of Series according to input correspondence.
+
+        Used for substituting each value in a Series with another value,
+        that may be derived from a function, a ``dict`` or
+        a :class:`Series`.
+
+        Parameters
+        ----------
+        arg : function, collections.abc.Mapping subclass or Series
+            Mapping correspondence.
+        na_action : {None, 'ignore'}, default None
+            If 'ignore', propagate NaN values, without passing them to the
+            mapping correspondence.
+
+        Returns
+        -------
+        Series
+            Same index as caller.
+
+        Examples
+        --------
+        >>> s = cudf.Series(['cat', 'dog', np.nan, 'rabbit'])
+        >>> s
+        0      cat
+        1      dog
+        2      NaN
+        3   rabbit
+        dtype: object
+
+        ``map`` accepts a ``dict`` or a ``Series``. Values that are not found
+        in the ``dict`` are converted to ``NaN``, unless the dict has a default
+        value (e.g. ``defaultdict``):
```

Comment:
We don't currently support `defaultdict`-like objects here. Doing so would be a bit non-trivial where I'd suggest we:
- update the docstring for now indicating default values in dicts aren't supported yet
- throw a `NotImplementedError` if the `__missing__` attribute is defined for the passed in dict
- raise a github issue about supporting `defaultdict`-like objects

Reply:
Awesome! Will do!

Reply:
Added an issue for this! It is issue #6724